### ISSUE TITLE
REST API should set the `owner` of all resources to the authenticated user

### ISSUE THREAD

**Issue by jonrkarr** [reproducible-biomedical-modeling/Biosimulations#438]
1. I logged into the REST API docs at https://biosimulations-api.herokuapp.com/docs with my GitHub account
2. I tried to create and then edit two models (ids: `test` and `jonrkarr-test`)
3. The models were created (they can retrieved with `​/models​/{identifier}`)
4. The owner is bilalshaikh42
5. I cannot edit these models
   Request: 
   ```
   curl -X PUT "https://biosimulations-api.herokuapp.com/models/jonrkarr-test" -H "accept: application/json" -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IlJqRkNPVGd4UXpNNU5UaEROa1U1TkVGRlJEUXlSalF5UlVJek5UbERSakEzUlRnNFJUWXhSZyJ9.eyJpc3MiOiJodHRwczovL2NyYm0uYXV0aDAuY29tLyIsInN1YiI6Imdvb2dsZS1vYXV0aDJ8MTE3NTMxNjQxOTc2MDM2MTgxNTczIiwiYXVkIjoiYXBpLmJpb3NpbXVsYXRpb25zLm9yZyIsImlhdCI6MTU4NjE0NDA0OSwiZXhwIjoxNTg2MTUxMjQ5LCJhenAiOiJXRVBVTWIySm8yOE5kRXQxWjdmaFV4NTRCZmY4TW5LRiIsInNjb3BlIjoiIn0.EhAk-os0JIdtDaDWL0lUeuhSAI9cyhdCIV2TXOhBZX_t9M6nK5BKCqrAmEPXCvACXMK5c8ZVajDa68iyO6NNLFkhiOn5WILZkozFNDThW8IH87uhE4yPZ6dYrQc1bIMCKv4L_Wb9pfGAYTZb6HBOZg3jXBkPH58u4P-yt3c_u5Qn7PC1hnAWF-qCIC09mRuvT7xJ9L2GQc6_wX9HTTAaAa5vLNNRLzhTLNz1fNkiLq0aj5apZsVKF12Y22yFlYlw62VwOuXFW8VPNg3TsJ77MZs0ya2M8c7pstD5TYK1PsqWZHeW4cWlMId9yrtr00kP3cTLJrbEA4NHfpo8qyeynw" -H "Content-Type: application/json" -d "{\"created\":\"2020-04-06T03:36:42.513Z\",\"id\":\"jonrkarr-test\",\"name\":\"test name\"}"
   ```

   Response:
   ```
   Not Authorized to edit Model jonrkarr-test
   ```

**Comment by bilalshaikh42** [comment]
It seems your test account did not create a new user in the database. I am not sure why this happened. Did you log into the front end with the account by any chance? If not, then it might not have sent the necessary info needed for the creation of a new user

**Comment by jonrkarr** [comment]
I figured out the problem. If the owner is not set in the POST creation requests, the default owner (bilalshaikh42) is used.

We need to change a few things:
- Neither the resource creation web forms nor the REST API should accept and `owner` property.
- The `owner` property should be automatically set based on the logged in user. 
- Similarly, the PUT requests should not accept an `owner` property. Users should not be able to change the owner of a resource.
- Similarly, the POST and PUT requests should not accept `created` and `updated` arguments. These should be automatically set by the API (which appears to be the case).
- As a result, we need to define slightly different schemas for the POST/PUT and GET requests

**Comment by bilalshaikh42** [comment]
A few thoughts on this
1) Ive implemented this with the idea that owners can be changed, simmilar to github changing repositories owner
2) Owner should automatically be set using the auth header if the provided owner is null. If not, that is bug 
3) Agreed on the put and post options not taking create and updated. The need for slightly different schemas for each request was a large motivation to using typescript to avoid rewriting it repeatedly. Currently, even though the OAS schema is asking for those fields, they will be ignored by the implementation

 I will update the new repo to implement the rest of the API and start making these changes

**Comment by jonrkarr** [comment]
(a) I see. Allowing users to change the owner is fine, except there's no parallel concept to an organization yet. Users should only be allowed to transfer ownership to/from organizations that they belong to.

(b) When the `owner` property is null, the `owner` should be set to the currently authenticated user (user associated with the authorization header). Currently, if `owner` is null, `owner` is set to `bilalshaikh42`. It sounds like there's a bug somewhere in the setting of the owner or getting the user associated with the authorization header.

**Comment by bilalshaikh42** [comment]
Yup, b is supposed to happen as you described, but if it is not then there
is a bug. I'll work on it.

Best
Bilal

**Comment by bilalshaikh42** [comment]
To be addressed in #461
